python/metatensor-operations/metatensor/operations/unique_metadata.py

```diff
@@ -99,19 +104,24 @@ def unique_metadata(
         object has len(``names``) entries.
     """
     # Parse input args
-    if not isinstance(tensor, TensorMap):
+    is_tensor_map = (
+        isinstance(tensor, TensorMap)
+        if torch_jit_is_scripting()
+        else check_isinstance(tensor, TensorMap)
+    )
+    if not is_tensor_map:
```

Comment:
Same as other PR, this should be 

```suggestion
    if not torch_jit_is_scripting() and not check_isinstance(tensor, TensorMap):
```